hw/ip/otbn/doc/_index.md

```diff
@@ -439,6 +439,25 @@ outer_body:
 
 ## Design Details
 
+### Memories
+
+The OTBN processor core has access to two dedicated memories:
+an instruction memory (IMEM), and a data memory (DMEM).
+Each memory is 4 kiB in size.
+
+The memory layout follows the Harvard architecture.
+Both memories are byte-addressed, with addresses starting at 0.
+
+The instruction memory (IMEM) is 32b wide and provides the instruction stream to the OTBN processor;
+it cannot be read or written from user code through load or store instructions.
+
+The data memory (DMEM) is 256b wide and read-write accessible from the base and big number instruction subsets of the OTBN processor core.
+When accessed from the base instruction subset through the `LW` or `SW` instructions, accesses must read or write 32b-aligned 32b words.
+When accessed from the big number instruction subset through the `BN.LID` or `BN.SID` instructions, accesses must read or write 256b-aligned 256b words.
```

Comment:
Maybe it would make sense to say that this is an exhaustive list? Something like this, maybe?
```
There are four instructions that can access data memory.
In the base instruction subset, there are `LW` (load word) and `SW` (store word).
These access 32b-aligned 32b words.
In the big number instruction subset, there are `BN.LID` (load indirect) and `BN.SID` (store indirect).
These access 256b-aligned 256b words.
```

I know it's more verbose, but I think it might be more helpful? Also, maybe we should be linking instruction mnemonics to the ISA document? (It's a bit painful to do though: maybe we should make a hugo shortcode?)